packages/hardhat/test/e2e/periphery/controller-helper.ts

```diff
@@ -0,0 +1,455 @@
+      const scaledEthPrice = ethPrice.div(10000)
+      const debtInEth = mintRSqueethAmount.mul(scaledEthPrice).div(one)
+      const collateralAmount = debtInEth.mul(3).div(2).add(ethers.utils.parseUnits('0.01'))
+      const tokenIndexAfter = await (positionManager as INonfungiblePositionManager).totalSupply();
+      const tokenId = await (positionManager as INonfungiblePositionManager).tokenByIndex(tokenIndexAfter.sub(1));
+
+      await (positionManager as INonfungiblePositionManager).connect(depositor).approve(controller.address, tokenId)
+      await controller.connect(depositor).mintWPowerPerpAmount(0, mintWSqueethAmount, tokenId)
+    })
+
+    it("close LP position deposited as collateral in vault", async () => {
+      const vaultId = (await shortSqueeth.nextId()).sub(1);
+      const uniTokenId =  (await controller.vaults(vaultId)).NftCollateralId;
+      const vaultBefore = await controller.vaults(vaultId); 
+      const ethPrice = await oracle.getTwap(ethUsdcPool.address, weth.address, usdc.address, 420, true)
+      const scaledEthPrice = ethPrice.div(10000)
+      const debtInEth = vaultBefore.shortAmount.mul(scaledEthPrice).div(one)
+      const collateralToFlashloan = debtInEth.mul(3).div(2).add(ethers.utils.parseUnits('0.01'))
+      const squeethPrice = await oracle.getTwap(wSqueethPool.address, wSqueeth.address, weth.address, 420, true)
+      const slippage = BigNumber.from(3).mul(BigNumber.from(10).pow(16))
+      const limitPriceEthPerPowerPerp = squeethPrice.mul(one.add(slippage)).div(one);
+      const positionBefore = await (positionManager as INonfungiblePositionManager).positions(uniTokenId);
+      const flashloanCloseVaultLpNftParam = {
+        vaultId: vaultId,
+        tokenId: uniTokenId,
+        liquidity: positionBefore.liquidity,
+        liquidityPercentage: ethers.utils.parseUnits('1'),
+        wPowerPerpAmountToBurn: vaultBefore.shortAmount.toString(),
+        collateralToFlashloan: collateralToFlashloan.toString(),
+        collateralToWithdraw: BigNumber.from(0),
+        limitPriceEthPerPowerPerp: limitPriceEthPerPowerPerp.toString(),
+        amount0Min: 0,
+        amount1Min: 0
+      }
+
+      await controller.connect(depositor).updateOperator(vaultId, controllerHelper.address);
+      await controllerHelper.connect(depositor).flashloanCloseVaultLpNft(flashloanCloseVaultLpNftParam, {value: ethers.utils.parseUnits('2')});
+
+      const positionAfter = await (positionManager as INonfungiblePositionManager).positions(uniTokenId);
+      const vaultAfter = await controller.vaults(vaultId); 
+
+      expect(positionAfter.tickLower === -887220).to.be.true
+      expect(positionAfter.tickUpper === 887220).to.be.true
+      expect(positionAfter.liquidity.eq(BigNumber.from(0))).to.be.true
+      expect(vaultAfter.shortAmount.eq(BigNumber.from(0))).to.be.true
+      expect(vaultAfter.collateralAmount.eq(BigNumber.from(0))).to.be.true
```

Comment:
Still think can improve these tests - For example, did the user get out the correct amt of ETH in their wallet?